### PR TITLE
[ISSUE-732] Pass reservation controller variables

### PR DESCRIPTION
**Pull request by safronovD**
### Issue dell/csi-baremetal#732

Related changes with https://github.com/dell/csi-baremetal/pull/730

## PR checklist
- [x] Add link to the issue
- [x] Choose Project
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
```
user@user-vm ~/g/s/g/d/csi-baremetal-operator (bugfix-test-long-reconcile-on-openshift)> 
helm install csi-baremetal ./charts/csi-baremetal-deployment/ --set image.tag=1.1.0-560.ea660fd --set image.pullPolicy=IfNotPresent --set scheduler.patcher.enable=true --set driver.drivemgr.type=loopbackmgr --set driver.drivemgr.deployConfig=true --set driver.log.level=debug --set scheduler.log.level=debug --set nodeController.log.level=debug --set driver.controller.reservation.slowTimeout=30s
```
```
    Environment:
      POD_IP:                     (v1:status.podIP)
      CSI_ENDPOINT:              unix:///csi/csi.sock
      LOG_FORMAT:                text
      KUBE_NODE_NAME:             (v1:spec.nodeName)
      NAMESPACE:                 default (v1:metadata.namespace)
      RESERVATION_FAST_TIMEOUT:  1500ms
      RESERVATION_SLOW_TIMEOUT:  30s
      RESERVATION_ATTEMPTS_NUM:  30
```

### CODE REVIEW

File: api/v1/components/controller.go [hunk 1]
```diff
@@ -23,5 +23,8 @@ type Controller struct {
 	Sidecars map[string]*Sidecar `json:"sidecars,omitempty"`
 	// +nullable
 	// +optional
-	Resources *ResourceRequirements `json:"resources,omitempty"`
+	Resources   *ResourceRequirements `json:"resources,omitempty"`
+	FastTimeout string                `json:"fastTimeout,omitempty"`
```
Comment (rkiselenko) [inline api/v1/components/controller.go:27]: There is a naming convention. https://github.com/zecke/Kubernetes/blob/master/docs/devel/api-conventions.md#naming-conventions

File: charts/csi-baremetal-operator/crds/csi-baremetal.dell.com_deployments.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.7.0
```
Comment (rkiselenko) [inline charts/csi-baremetal-operator/crds/csi-baremetal.dell.com_deployments.yaml:7]: Why version changed?

File: pkg/controller.go [hunk 3]
```diff
@@ -151,6 +156,9 @@ func createControllerContainers(csi *csibaremetalv1.Deployment) []corev1.Contain
 				{Name: "NAMESPACE", ValueFrom: &corev1.EnvVarSource{
 					FieldRef: &corev1.ObjectFieldSelector{APIVersion: "v1", FieldPath: "metadata.namespace"},
 				}},
+				{Name: fastTimeoutEnv, Value: c.FastTimeout},
+				{Name: slowTimeoutEnv, Value: c.SlowTimeout},
+				{Name: attemptsNumEnv, Value: strconv.Itoa(int(c.MaxAttempts))},
```
Comment (rkiselenko) [inline pkg/controller.go:161]: Double conversion? `string -> int -> string` ?

**Review by rkiselenko — CHANGES_REQUESTED** [review]

File: api/v1/components/controller.go (hunk 1 above)
Comment (rkiselenko) [inline api/v1/components/controller.go:27]: Add comments above every field, so `CRD` will generate documentation.

**Review by rkiselenko — CHANGES_REQUESTED** [review]

File: pkg/controller.go [hunk 4]
```diff
@@ -31,6 +31,11 @@ const (
 	healthPort = 9999
 
 	provisionerTimeout = "30s"
+
+	// reservation parameters
+	fastTimeoutEnv = "RESERVATION_FAST_TIMEOUT"
```
Comment (rkiselenko) [inline pkg/controller.go:36]: Why that's strings in own variables ?

**Review by rkiselenko — CHANGES_REQUESTED** [review]

File: api/v1/components/controller.go (hunk 1 above)
Comment (safronovD) [reply api/v1/components/controller.go]: Refactored with RateLimiter args names from [go-client](https://github.com/kubernetes/client-go/blob/8f44946f6cbe967fbe2e2548e76987680a89428e/util/workqueue/default_rate_limiters.go#L136)

File: charts/csi-baremetal-operator/crds/csi-baremetal.dell.com_deployments.yaml (hunk 2 above)
Comment (safronovD) [reply charts/csi-baremetal-operator/crds/csi-baremetal.dell.com_deployments.yaml:7]: Fixed

File: pkg/controller.go (hunk 3 above)
Comment (safronovD) [reply pkg/controller.go]: No, `MaxAttempts` is `uint`

File: api/v1/components/controller.go (hunk 1 above)
Comment (safronovD) [reply api/v1/components/controller.go]: Comments added

File: pkg/controller.go (hunk 4 above)
Comment (safronovD) [reply pkg/controller.go]: What do you mean? They are constants

File: pkg/controller.go (hunk 3 above)
Comment (rkiselenko) [reply pkg/controller.go]: Why not `strconv.FormatUint` ?

File: pkg/controller.go (hunk 4 above)
Comment (rkiselenko) [reply pkg/controller.go]: What the point of this constants ? Why this variables are in constants and for example `NAMESPACE` not ?

File: pkg/controller.go (hunk 3 above)
Comment (safronovD) [reply pkg/controller.go]: Fixed

**Review by mishoyama — APPROVED** [review]

File: pkg/controller.go (hunk 4 above)
Comment (safronovD) [reply pkg/controller.go]: To clarify names of environment variables in a pod

Comment (safronovD) [reply pkg/controller.go]: But I am not sure that I understand what we are talking about. What's up with Namespace here? It's passed from pod spec

Comment (rkiselenko) [reply pkg/controller.go]: The main feature of constants is to share the same value and make code more DRY, you decide to move that strings in variable, what the point ? Does that strings also used somewhere in the code several times?

**Review by rkiselenko — APPROVED** [review]

File: pkg/controller.go (hunk 4 above)
Comment (safronovD) [reply pkg/controller.go]: To not use long string literals in the plain code is the default practice. They are used in csi-baremetal repo in controller also, so we can easily verify that they are the same

**Pull request merged**